src/Symfony/Bundle/FrameworkBundle/Routing/Router.php

```diff
@@ -130,6 +138,10 @@ private function resolveParameters(RouteCollection $collection)
      */
     private function resolve($value)
     {
+        if (!$paramFetcher = $this->paramFetcher) {
```

Comment:
shouldn't we throw instead in the constructor to prevent dealing with this case here?

Reply:
One could theoretically use this router without the need to resolve parameters, so instead we could rather move up this check in `getRouteCollection` to never call `resolveParameters` on no way to retrieve parameters.
But yes for now the exception looks fine to me. Updated.